### PR TITLE
Fix of issue NuGet/Home#1802: Pre select packages in consolidated tab.

### PR DESCRIPTION
**Pull request by feiling**
In fact, with this fix, we do auto select in both the consolidated and the updates tabs.

@yishaigalatzer @emgarten @deepakaravindr

### CODE REVIEW

File: src/NuGet.Clients/PackageManagement.UI/PackageSolutionDetailControlModel.cs [hunk 1]
```diff
@@ -412,6 +413,34 @@ private static bool IsInstalled(NuGetProject project, string id)
...
+                _filter == Filter.UpdatesAvailable)
+            {
+                foreach (var project in _projects)
+                {
+                    project.IsSelected = project.InstalledVersion != null;
+                }
+            }
+        }
+
+        public override void OnFilterChanged(Filter filter)
```
Comment (feiling) [inline src/NuGet.Clients/PackageManagement.UI/PackageSolutionDetailControlModel.cs]: we need to clear selections when the current tab is switched to "Browser" or "install".

**Comment by emgarten** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
@feiling I'd like to see a short demo

File: src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs [hunk 2]
```diff
@@ -200,4 +201,17 @@ public void SelectFilter(Filter selectedFilter)
...
     }
+
+    public class FilterChangedEventArgs : EventArgs
+    {
+        public Filter? PreviousFilter
+        {
+            get;
+        }
+
+        public FilterChangedEventArgs(Filter? previousFilter)
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs:212]: This confused me a bit, we could have used a Filter.None instead, but I guess this works and might actually be cleaner.

**Comment by yishaigalatzer** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
@feiling does selection change after we install a package say from powershell console?

**Comment by feiling** [comment]
Unfortunately no. Actions in powershell won't cause UI to refresh. This is a known issue.

**Comment by feiling** [comment]
Fix merged.

**Pull request closed without merging**